jobs/jobs/pipeline.py

```diff
@@ -0,0 +1,47 @@
+from abc import ABCMeta, abstractmethod
+from dataclasses import dataclass
+from typing import List, Optional
+
+
+@dataclass
+class PipelineFileInput:
+    job_id: int
+
+
+@dataclass
+class PipelineFile:
+    bucket: str
+    input: PipelineFileInput
+    input_path: str
+    pages: List[int]
+    output_path: Optional[str] = None
+    s3_signed_url: Optional[str] = None
+    annotation_id: Optional[str] = None
+
+
+@dataclass
+class PipelineRunArgs:
+    job_id: int
+    tenant: str
+    files_data: List[PipelineFile]
+
+
+@dataclass
+class AnyPipeline:
+    pass
+
+
+class BasePipeline(metaclass=ABCMeta):
+    @abstractmethod
+    async def list(self) -> List[AnyPipeline]:
+        raise NotImplementedError()
```

Comment:
You can skip this as @abstractmethod takes care of it
```suggestion
        pass
```

Reply:
The `@abstractmethod` doesn't cover all scenarios of using an abstract class. For instance,
```
BasePipeline.list(AirflowPipeline())
```
with `pass` in the `list` method won't raise an exception.

While this example might seem odd, it's important to prevent improper usage of the created class hierarchy.